examples/common/python/connectors/direct/tcs_listener/tcs_worker_registry_handler.py

```diff
@@ -184,62 +183,47 @@ def __process_worker_set_status(self, worker_id, input_json_str, response):
         return response
 # ------------------------------------------------------------------------------------------------
 
-    def __lookup_basic(self, lookup_bool, input_json_str, response):
-        work_orders = []
-        work_orders = self.kv_helper.lookup("workers")
-
-        alter_items = []
-        alter_items = utility.list_difference(self.worker_pool, work_orders)
-
-        for item in alter_items:
-            self.worker_pool.remove(item)
-
-        alter_items = utility.list_difference(work_orders, self.worker_pool)
-        for item in alter_items:
-            self.worker_pool.append(item)
+    def __lookup_basic(self, is_lookup_next, input_json_str, response):
+        # sync the work pool to that of DB
+        self.worker_pool = self.kv_helper.lookup("workers")
 
         input_value = json.loads(input_json_str)
         total_count = 0
         ids = []
         lookupTag = ""
 
+        params = input_value["params"]
         for worker_id in self.worker_pool:
-            if lookup_bool:
-                if (worker_id == input_value["params"]["lookUpTag"]):
-                    lookup_bool = False
-                    continue
-                else:
-                    continue
+            if is_lookup_next:
+                # loop until found the expected lookUpTag
+                is_lookup_next = (worker_id != params["lookUpTag"])
```

Comment:
On line 199 you use `lookUpTag`:
> ` 199 is_lookup_next = (worker_id != params["lookUpTag"])`
but on line 223 you use "`lookupTag`
> `223 "lookupTag": lookupTag,`
These should be the same.  Looking at other source files, `lookUpTag` is only used here, so change `lookUpTag` to `lookupTag`

Reply:
done